app/Actions/Photo/Create.php

```diff
@@ -118,6 +121,22 @@ public function add(
 			$res = $this->save($this->photo);
 		}
 
+		//extract the main color and palette
+		$palette = ColorThief::getPalette($this->path, 5);
+
+		if (!empty($palette)) {
+			foreach ($palette as $index => $value) {
+				$color = new Color([
+					'r' => $value[0],
+					'g' => $value[1],
+					'b' => $value[2],
+					'is_main' => ($index === 0 ? 1 : 0),
+					'photo_id' => $this->photo->id,
+				]);
+				$this->photo->colors()->save($color);
+			}
+		}
+
```

Comment:
It was just yesterday, when I learned the hard way, that you should unset a cached relation after one has modified it by `->make` or `->save` due to an Eloquent bug. Otherwise bad things might happen. Hence, after the `foreach`-loop when you have finished with all `$this->photo->colors()->save()`, you need to add

    if ($this->photo->relationLoaded('colors')) {
	  $this->photo->unsetRelation('colors');
    }